packages/peregrine/lib/talons/CreateAccount/useCreateAccount.js

```diff
@@ -1,24 +1,59 @@
-import { useMemo } from 'react';
+import { useCallback, useEffect, useMemo } from 'react';
+import { useLazyQuery } from '@apollo/react-hooks';
 import { useUserContext } from '@magento/peregrine/lib/context/user';
 
 /**
  * Returns props necessary to render CreateAccount component.
  *
  * @param {Object} props.initialValues initial values to sanitize and seed the form
  * @returns {{
- *   hasError: boolean,
  *   isDisabled: boolean,
  *   isSignedIn: boolean,
  *   initialValues: object
  * }}
  */
 export const useCreateAccount = props => {
-    const { initialValues = {} } = props;
+    const { initialValues = {}, onSubmit, query } = props;
 
     const [
         { createAccountError, isCreatingAccount, isSignedIn }
     ] = useUserContext();
-    const hasError = !!createAccountError;
+
+    const [runQuery, queryResponse] = useLazyQuery(query);
+    const { called, loading, error, data } = queryResponse;
+
+    const handleSubmit = useCallback(
+        values => {
+            runQuery({
+                variables: {
+                    email: values.customer.email
+                }
+            });
+        },
+        [runQuery]
+    );
+
+    useEffect(() => {
+        if (called && !loading && !error && data) {
+            if (data.isEmailAvailable.is_email_available) {
+                onSubmit();
+            }
+        }
+    }, [onSubmit, called, loading, error, data]);
+
+    // Mapping of message to type is done in the UI component.
+    const errors = [];
+    if (createAccountError) {
+        errors.push({
+            type: 'CREATE_ACCOUNT_ERROR'
+        });
+    }
+
+    if (data && !data.isEmailAvailable.is_email_available) {
```

Comment:
Would be! I might be able to rewrite the query.

Reply:
So I couldn't rewrite the query _but_ there is a `createCustomer` mutation which I may be able to use instead of the create account endpoint. I'm going to go down a rabbit hole of rewriting this again :D